### PR TITLE
use the logic in Object Store to update schema version

### LINKED ISSUE #3417: Use ObjectStore functions for creating meta tables
To ensure full and stable compatibility between platforms we need to use Object Store functionality for creating the `meta` and `pk` tables.

See e.g https://github.com/realm/realm-java/pull/3409#discussion-diff-78084779

### PR DESCRIPTION
**Pull request by zaki50**
fixes #3417

### CODE REVIEW

**Comment by zaki50** [comment]
ready for review @realm/java

File: realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp [hunk 1]
```diff
@@ -148,6 +148,24 @@ Java_io_realm_internal_SharedRealm_nativeGetVersion(JNIEnv *env, jclass, jlong s
     return -1;
 }
 
+JNIEXPORT void JNICALL
+Java_io_realm_internal_SharedRealm_nativeSetVersion(JNIEnv *env, jclass, jlong shared_realm_ptr, jlong version)
```
Comment (beeender) [inline realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp]: Rename to `nativeSetSchemaVersion` instead since we have a method `nativeGetVersion` is to get the `SharedGroup` version.

File: realm/realm-library/src/main/java/io/realm/RealmSchema.java [hunk 2]
```diff
@@ -87,7 +87,7 @@ public RealmObjectSchema get(String className) {
         Set<RealmObjectSchema> schemas = new LinkedHashSet<>(tableCount);
         for (int i = 0; i < tableCount; i++) {
             String tableName = realm.sharedRealm.getTableName(i);
-            if (Table.isMetaTable(tableName)) {
+            if (!Table.isModelTable(tableName)) {
```
Comment (beeender) [inline realm/realm-library/src/main/java/io/realm/RealmSchema.java]: Is this correct? For the old logic, the loop will be skipped if it is a primary key table.

File: realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp (hunk 1 above)
Comment (zaki50) [inline realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp]: `nativeGetVersion()` seems to be also dealing with schema version.

```
JNIEXPORT jlong JNICALL
Java_io_realm_internal_SharedRealm_nativeGetVersion(JNIEnv *env, jclass, jlong shared_realm_ptr)
{
    TR_ENTER_PTR(env, shared_realm_ptr)

    auto shared_realm = *(reinterpret_cast<SharedRealm*>(shared_realm_ptr));
    try {
        return static_cast<jlong>(ObjectStore::get_schema_version(shared_realm->read_group()));
    } CATCH_STD()

    // FIXME: Use constant value
    return -1;
}
```

File: realm/realm-library/src/main/java/io/realm/RealmSchema.java (hunk 2 above)
Comment (zaki50) [inline realm/realm-library/src/main/java/io/realm/RealmSchema.java]: I think modified code also skips primary key table since primary key table does not have `class_` prefix.
Do I have something overlooked?

File: realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp (hunk 1 above)
Comment (beeender) [inline realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp]: Uh, the other one is `Java_io_realm_internal_SharedRealm_nativeGetVersionID()` ... Sorry... Maybe change them both to `SchemaVersion()`? But let's do it in another PR ... they are not related with this one.

File: realm/realm-library/src/main/java/io/realm/RealmSchema.java (hunk 2 above)
Comment (beeender) [inline realm/realm-library/src/main/java/io/realm/RealmSchema.java]: Nevermind, my bad :)

**Comment by beeender** [comment]
:+1:

**Comment by kneth** [comment]
:+1:
